### PR TITLE
GitHub actions

### PR DESCRIPTION
**Pull request by rishj2612**

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+        registry-url: 'https://registry.npmjs.org'
+    - name: Install dependencies    
+      run: npm ci
+    - name: Build  
+      run: npm run build --if-present
+    - name: Prettier
```
Comment (shukla2512) [inline .github/workflows/pr.yml]: We don't need this.

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    - uses: actions/checkout@v3
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+        registry-url: 'https://registry.npmjs.org'
+    - name: Install dependencies    
+      run: npm ci
+    - name: Build  
+      run: npm run build --if-present
```
Comment (shukla2512) [inline .github/workflows/pr.yml]: --if--present ?

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+        registry-url: 'https://registry.npmjs.org'
+    - name: Install dependencies    
+      run: npm ci
+    - name: Build  
+      run: npm run build --if-present
+    - name: Prettier
```
Comment (shukla2512) [inline .github/workflows/publish.yml]: No need of this.

File: .github/workflows/pr.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+name: Validate PR
```
Comment (shukla2512) [inline .github/workflows/pr.yml]: Just thinking if the action name should be 
Build and Test. - Runs for all branches.
Publish - Runs for main.

File: .github/workflows/pr.yml (hunk 1 above)
Comment (rishj2612) [reply .github/workflows/pr.yml]: removed prettier

File: .github/workflows/pr.yml (hunk 4 above)
Comment (rishj2612) [reply .github/workflows/pr.yml]: updated the name

File: .github/workflows/pr.yml (hunk 2 above)
Comment (rishj2612) [reply .github/workflows/pr.yml]: removed it

**Pull request merged**